### PR TITLE
workflows: update archive action

### PR DESCRIPTION
**Pull request by phlogistonjohn**
When archiving, use the latest version of the archive action and use
it's ability to archive an entire directory.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -45,17 +45,12 @@ jobs:
...
-        path: "_results/coverage/go-ceph.html"
-    - name: Archive implements results
-      if: "matrix.ceph_version == 'pacific'"
-      uses: actions/upload-artifact@v2
-      with:
-        name: "go-ceph-implements-${{ matrix.ceph_version }}"
-        path: "_results/implements.*"
+        name: "go-ceph-results-${{ matrix.ceph_version }}"
+        path: |
+          _results/
```
Comment (anoopcs9) [inline .github/workflows/main.yml:53]: `path: _results/` also works, right?

Comment (phlogistonjohn) [reply .github/workflows/main.yml:53]: It should. The newer version can support multiple paths and exclusions and so the docs use the "|" line continuation syntax in some places. I changed it to partly serve as a reminder those options now exist.

Comment (anoopcs9) [reply .github/workflows/main.yml:53]: Ok.

**Comment by anoopcs9** [comment]
How about defining some value for `retention days`?

**Comment by phlogistonjohn** [comment]
> How about defining some value for `retention days`?

Sounds good. Will do. I think I will set it to one week unless you feel otherwise.

**Comment by ansiwen** [comment]
> > How about defining some value for `retention days`?
> 
> Sounds good. Will do. I think I will set it to one week unless you feel otherwise.

Isn't that a bit short? I had a similar situation in another project recently, where all artifacts have been removed after only a week, and that was really disturbing, especially when you want to reproduce something from earlier. I would go with one month.

**Comment by phlogistonjohn** [comment]
> I would go with one month.

Works for me.

**Review by anoopcs9 — APPROVED** [review]
lgtm, thanks.

**Pull request merged**